File: backend/agentflow/schemas/knowledge.py
"""Knowledge base request/response schemas."""
from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, Field


class DatasetCreate(BaseModel):
    name: str
    description: str = ""
    index_method: Literal["high_quality", "economy"] = "high_quality"


class RetrieveRequest(BaseModel):
    query: str
    top_k: int = 3
    score_threshold: float = 0.5
    search_mode: Literal["semantic", "keyword", "hybrid"] = "hybrid"
    rerank_enabled: bool = False
    semantic_weight: float = 0.7
    keyword_weight: float = 0.3


class RetrieveResult(BaseModel):
    content: str
    score: float
    metadata: dict[str, Any] = {}

File: frontend/src/types/knowledge.ts
export interface Dataset {
  id: string
  name: string
  description: string
  index_method: 'high_quality' | 'economy'
  embedding_model: string
  created_at?: string
}

export interface KnowledgeDocument {
  id: string
  dataset_id: string
  name: string
  chunk_strategy: 'general' | 'parent_child' | 'qa'
  status: 'pending' | 'indexing' | 'done' | 'failed'
  chunk_count: number
  created_at?: string
}

export interface RetrieveResult {
  content: string
  score: number
  metadata: Record<string, any>
}

export interface RetrievalConfig {
  query: string
  top_k: number
  score_threshold: number
  search_mode: 'semantic' | 'keyword' | 'hybrid'
  rerank_enabled: boolean
  semantic_weight: number
  keyword_weight: number
}
